### PR TITLE
fix: add missing goreleaser ldflags

### LINKED ISSUE #2725: `/health` reports `0.0.0` for api versions
Using Bee-factory images that uses Bee docker images I am getting:

```
$ curl http://localhost:1635/health
{"status":"ok","version":"1.4.1-238867f1","apiVersion":"0.0.0","debugApiVersion":"0.0.0"}
```

### PR DESCRIPTION
**Pull request by mrekucci**
### Checklist

- [ ] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md)
- [ ] My change requires a documentation update and I have done it
- [ ] I have added tests to cover my changes.

### Screenshots (if appropriate):

---
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2726)

Closes #2725

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -30,7 +37,14 @@ builds:
...
     ldflags:
-      - -s -w -X github.com/ethersphere/bee.version={{.Version}} -X github.com/ethersphere/bee.commitHash={{.ShortCommit}} -X github.com/ethersphere/bee.commitTime={{.CommitTimestamp}}
+      - -s -w
+      - -X github.com/ethersphere/bee.version={{.Version}}
+      - -X github.com/ethersphere/bee.commitHash={{.ShortCommit}}
+      - -X github.com/ethersphere/bee.commitTime={{.CommitTimestamp}}
+      - -X github.com/ethersphere/bee/pkg/api.Version={{.BeeApiVersion}}
+      - -X github.com/ethersphere/bee/pkg/debugapi.Version={{.BeeDebugApiVersion}}
+      - -X github.com/ethersphere/bee/pkg/p2p/libp2p.reachabilityOverridePublic={{.ReachabilityOverridePublic}}
+      - -X github.com/ethersphere/bee/pkg/postage/listener.batchFactorOverridePublic={{.BatchfactorOverridePublic}}
```
Comment (acud) [inline .goreleaser.yml:47]: I would explicitly not set this

File: .goreleaser.yml [hunk 2]
```diff
@@ -30,7 +37,14 @@ builds:
...
       - -trimpath
     ldflags:
-      - -s -w -X github.com/ethersphere/bee.version={{.Version}} -X github.com/ethersphere/bee.commitHash={{.ShortCommit}} -X github.com/ethersphere/bee.commitTime={{.CommitTimestamp}}
+      - -s -w
+      - -X github.com/ethersphere/bee.version={{.Version}}
+      - -X github.com/ethersphere/bee.commitHash={{.ShortCommit}}
+      - -X github.com/ethersphere/bee.commitTime={{.CommitTimestamp}}
+      - -X github.com/ethersphere/bee/pkg/api.Version={{.BeeApiVersion}}
+      - -X github.com/ethersphere/bee/pkg/debugapi.Version={{.BeeDebugApiVersion}}
+      - -X github.com/ethersphere/bee/pkg/p2p/libp2p.reachabilityOverridePublic={{.ReachabilityOverridePublic}}
```
Comment (acud) [inline .goreleaser.yml:46]: also this. the build process should not even be aware we are providing these overrides

File: .goreleaser.yml (hunk 1 above)
Comment (mrekucci) [reply .goreleaser.yml]: Removed.

File: .goreleaser.yml (hunk 2 above)
Comment (mrekucci) [reply .goreleaser.yml]: Removed.

**Review by acud — APPROVED** [review]
Thanks! Does this need to get wired in somehow now? @vandot ?

File: .goreleaser.yml (hunk 2 above)
Comment (vandot) [reply .goreleaser.yml]: @acud they should be inside [Makefile](https://github.com/ethersphere/bee/blame/38787a97c52fc03de6f521caff552964bb4f9496/Makefile#L32)?

**Comment by vandot** [comment]
@mrekucci @acud if there is nothing to change in regards to [this](https://github.com/ethersphere/bee/pull/2726#discussion_r766514594) looking good, AFAIK `ldflags` are inside `.goreleaser.yaml` and `Makefile` nothing more to change

**Review by vandot — APPROVED** [review]

**Comment by mrekucci** [comment]
> @mrekucci @acud if there is nothing to change in regards to [this](https://github.com/ethersphere/bee/pull/2726#discussion_r766514594) looking good, AFAIK `ldflags` are inside `.goreleaser.yaml` and `Makefile` nothing more to change

How will be the `{{.BeeApiVersion}}` and {{.BeeDebugApiVersion}} set?

**Comment by vandot** [comment]
Aaaaa yes, they are set with this ugly [hack](https://github.com/ethersphere/bee/blob/38787a97c52fc03de6f521caff552964bb4f9496/Makefile#L21)
It looks to me that GoReleaser doesn't have support to load those variables from inside `.goreleaser.yaml` file so we have to do it outside, [here](https://github.com/ethersphere/bee/blob/38787a97c52fc03de6f521caff552964bb4f9496/.github/workflows/release.yaml#L49)

**Pull request merged**